doc/specs/#1502 - Advanced Tab Switcher/spec.md

```diff
@@ -0,0 +1,219 @@
+
+### Compatibility
+
+- The existing way of navigating horizontally through the tabs on the tab bar should not break.
+- These should also be separate keybindings from the keybindings associated with using the tab switcher.
+- Reordering tabs on the tab bar shouldn't change the MRU order. The MRU is essentially supposed to be a different _view_ of the already existing tab list.
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+We'll need to be careful about how the UI is presented depending on different sizes of the terminal. We also should test how the UI looks as it's open and resizing is happening. Visual Studio's tab switcher is a fixed size, and is always in the middle. Even when the VS window is smaller than the tab switcher size, the tab switcher will show up larger than the VS window itself.
+
+![Small Visual Studio Without Tab Switcher](img/VSMinimumSize.png)
+![Small Visual Studio With Tab Switcher](img/VSMinimumSizeWithTabSwitcher.png)
+
+Visual Studio Code only allows the user to shrink the window until it hits a minimum width and height. This minimum width and height gives its tab switcher enough space to show a meaningful amount of information.
+
+![Small Visual Studio Code with Tab Switcher](img/VSCodeMinimumTabSwitcherSize.png)
+
+Terminal can't really replicate Visual Studio's version of the tab switcher in this situation. The TabSwitcher needs to be contained within the Terminal. So, if the TabSwitcher is always centered and has a percentage padding from the borders of the Terminal, it'll shrink as Terminal shrinks. One thing I'm not too sure about is when the Terminal window is so small that not even a single tab title will be visible. Should it just not display the UI? Should it try to display the UI and whatever is visible, is visible?
+
+## Future considerations
+
+### Pane Navigation
+
+@zadiji-msft in [#1502] brought up the idea of pane navigation, inspired by tmux.
+
+![Tmux Tab and Pane Switching](img/tmuxPaneSwitching.png)
+
+Tmux allows the user to navigate directly to a pane and even give a preview of the pane. This would be extremely useful since it would allow the user to see a tree of their open tabs and panes. Currently there's no way to see what panes are open in each tab, so if you're looking for a particular pane, you'd need to cycle through your tabs to find it. If something like pane profile names (not sure what information to present in the switcher for panes) were presented in the TabSwitcher, the user could see all the panes in one box.
+
+To support pane navigation, the tab switcher can simply have another column to the right of the tab list to show a list of panes inside the selected tab. As the user iterates through the tab list, they can simply hit right to dig deeper into the tab's panes, and hit left to come back to the tab list.
+
+Question: Does MRU make sense for panes?

```

Comment:
Sure.

Reply:
Cool, then I'm thinking each list of Panes within a tab will be in MRU/In-Order, depending on what keybinding the user opened the Switcher with.